### PR TITLE
Icalendar export enhancements, new Ical serializers

### LINKED ISSUE #4785: iCalendar export: Do not emit Location property when there is no location
At https://indico.geant.org/event/8/ I click on “Download: Detailed Timetable calendar file”.  It contains LOCATION property with empty value.
```
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//CERN//INDICO//EN
BEGIN:VEVENT
SUMMARY:Life of BRIAN
DTSTART;VALUE=DATE-TIME:20210622T094500Z
DTEND;VALUE=DATE-TIME:20210622T100500Z
DTSTAMP;VALUE=DATE-TIME:20210217T085250Z
UID:indico-contribution-8-198@indico.geant.org
DESCRIPTION:Speakers: Erik Reid (GÉANT)\nGÉANT has for many years used a
  growing stack of legacy and bespoke software components for doing histori
 cal network analytics and high-level information sharing. This presentatio
 n describes the motivation and path from that legacy stack to a new and mo
 re maintainable system called BRIAN (Backbone Router Interface ANalytics).
 \n\nhttps://indico.geant.org/event/8/contributions/198/
LOCATION:
URL:https://indico.geant.org/event/8/contributions/198/
END:VEVENT
BEGIN:VEVENT
SUMMARY:Establishment of new identity federations in Africa: the AfricaCon
 nect3 perspective
...
```

### LINKED ISSUE #4786: Include contact data in iCalendar export
Shows https://indico.geant.org/event/8/ contact data: email and telephone number.  I export the event as iCalendar.  Neither the detailed nor the compressed data for the event contains the contact data.

* Either put the email as ORGANIZER, or as ATTENDEE property, and the telephone as CN property, or alter the generated text and include the information there, or put the contact data in the unstructured [CONTACT property](https://tools.ietf.org/html/rfc5545#section-3.8.4.2).

### LINKED ISSUE #4787: Include Logo and Color in iCalendar export
My reading of the code is that indico/modules/events/ical.py converts the event data to iCalendar .  The manual https://learn.getindico.io/conferences/custom_layout/ says that events can have logo and color.

Please extend the iCalendar export to include:
* the Logo, ideally as [IMAGE;VALUE=URI](https://tools.ietf.org/html/rfc7986#section-5.10) property, but alternatively as ATTACHMENT
* the [COLOR](https://tools.ietf.org/html/rfc7986#section-5.9) property.

For the latter, the RFC says this must be CSS3 name, but to my knowledge any software understands colour also as RGB value. In any case https://gitlab.gnome.org/GNOME/evolution/-/commit/736a5a3251a553de8f demonstrates how to convert RGB to the closest CSS3 color name.

### LINKED ISSUE #4791: Utilize RELATED-TO in iCalendar export for sub-events
I download https://indico.geant.org/export/categ/0.ics to retrieve all public data, as I was told at https://github.com/indico/indico/issues/4782#issuecomment-780172883.  Then I go to https://indico.geant.org/event/8/ , from there get the link https://indico.geant.org/event/8/event.ics?detail=contributions and using this URL I get data, that goes beyond all public data.

* What URL shall I use to get really all public data?
* Include in each “contributions”-subevent the property [RELATED-TO](https://tools.ietf.org/html/rfc5545#section-3.8.4.5) e.g. `RELATED-TO:indico-event-8@indico.geant.org`.

### PR DESCRIPTION
**Pull request by javfg**
Closes #4785, closes #4786, closes #4787, and closes #4791.

Also:
  * Includes options to download detailed timetable for Sessions, so UX is uniform with event ical download.
  * New serializer for Sessions and Contributions, similar to the one for Events, replacing the legacy ones.

### CODE REVIEW

File: indico/modules/events/sessions/ical.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from indico.core.config import config
+from indico.modules.events.ical import generate_basic_component
+from indico.web.flask.util import url_for
+
+
+def generate_session_component(session, related_to_uid=None):
+    """Generates an Event icalendar component from an Indico Session.
+
+    :param session: The Indico Session to use
+    :param related_to_uid: Indico uid used in related_to field
```
Comment (ThiefMaster) [inline indico/modules/events/sessions/ical.py:20]: What's an "Indico uid"? Maybe "iCal uid" would be clearer?

File: indico/modules/events/sessions/ical.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    :returns: an icalendar Event
+    """
+
+    uid = 'indico-session-{}@{}'.format(session.id, url_parse(config.BASE_URL).host)
```
Comment (ThiefMaster) [inline indico/modules/events/sessions/ical.py:24]: Could use an f-string here:

```suggestion
    uid = f'indico-session-{session.id}@{url_parse(config.BASE_URL).host}'
```

File: indico/modules/events/sessions/ical.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    if related_to_uid:
+        component.add('related_to', related_to_uid)
+
+    return component
+
+
+def session_to_ical(session, detail_level='sessions'):
+    """Serialize a contribution into an ical.
+
+    :param event: The contribution to serialize
```
Comment (ThiefMaster) [inline indico/modules/events/sessions/ical.py:37]: that doesn't look correct ;) (better to not have param docs at all than having wrong docs, so just omitting it would probably be fine since the argument name is)

File: indico/modules/events/sessions/ical.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    """
+
+    calendar = Calendar()
+    calendar.add('version', '2.0')
+    calendar.add('prodid', '-//CERN//INDICO//EN')
+
+    related_event_uid = 'indico-event-{}@{}'.format(session.event.id, url_parse(config.BASE_URL).host)
+
+    if detail_level == 'sessions':
```
Comment (ThiefMaster) [inline indico/modules/events/sessions/ical.py:46]: We only have two detail levels: just the session or all its contributions. So maybe a bool like `use_contributions` would make more sense?

File: indico/modules/events/sessions/ical.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    elif detail_level == 'contributions':
+        from indico.modules.events.contributions.ical import generate_contribution_component
+        components = [generate_contribution_component(contribution, related_event_uid)
+                      for contribution in session.contributions]
+        for component in components:
+            calendar.add_component(component)
+
+    data = calendar.to_ical()
+
+    return data
```
Comment (ThiefMaster) [inline indico/modules/events/sessions/ical.py:58]:
```suggestion
    return calendar.to_ical()
```

File: indico/modules/events/sessions/templates/display/session_display.html [hunk 6]
```diff
@@ -59,14 +59,11 @@
...
                     <a href="{{ url_for('sessions.export_ics', sess) }}" class="i-button icon-calendar"></a>
                 {% else %}
-                    <div id="session-calendar-link" data-session-id="{{ sess.id }}"
-                         data-event-id="{{ sess.event_id }}"></div>
+                    <div id="session-calendar-link" data-session-id="{{ sess.id }}" data-event-id="{{ sess.event_id }}"
+                         data-session-contrib-count="{{ sess.contributions|length }}"-></div>
                 {% endif %}
             {%- endif %}
         </div>
-        {% if sess.start_dt -%}
```
Comment (ThiefMaster) [inline indico/modules/events/sessions/templates/display/session_display.html:67]: I think this check may need to stay. Exporting a poster session's contributions to ical doesn't make sense since they have no time.

File: indico/modules/events/static/offline.py [hunk 7]
```diff
@@ -366,7 +367,7 @@ def _get_session(self, session):
         pdf = get_session_timetable_pdf(session, tz=self._display_tz)
         self._add_pdf(session, 'sessions.export_session_timetable', pdf)
 
-        self._add_file(get_session_ical_file(session), 'sessions.export_ics', session)
+        self._add_file(session_to_ical(session), 'sessions.export_ics', session)
```
Comment (ThiefMaster) [inline indico/modules/events/static/offline.py:370]: Have you tested these changes? `_add_file` seems to handle both file objects and strings, but I'd like to be sure there are no problems e.g. with implicit str/bytes happening (or failing) somewhere..

File: indico/modules/events/ical.py [hunk 8]
```diff
@@ -12,72 +12,135 @@
...
+                else (entity.venue_name or entity.room_name))
+    if location:
+        component.add('location', location)
+
+    # speakers are located in a different variable in contributions
+    speaker_list = None
+    if hasattr(entity, 'person_links'):
+        speaker_list = entity.person_links
+    elif hasattr(entity, 'speakers'):
+        speaker_list = entity.speakers
```
Comment (ThiefMaster) [inline indico/modules/events/ical.py:56]: This won't work as you expect: contributions also have `person_links` (authors, speakers, etc.) so you need to check for speakers first.

If all entities used here have at least `person_links` you can just use this:

```python
speaker_list = getattr(entity, 'speakers', entity.person_links)
```

Otherwise just reverse the order of your if/elif.

File: indico/modules/events/ical.py [hunk 9]
```diff
@@ -12,72 +12,135 @@
...
+        component.add('image', f"{config.BASE_URL}{event.logo_url}", {'VALUE': 'URI'})
+
+    return component
+
+
+def event_to_ical(event, user=None, detail_level='events'):
     """Serialize an event into an ical.
 
     :param event: The event to serialize
     :param user: The user who needs to be able to access the events
```
Comment (ThiefMaster) [inline indico/modules/events/ical.py:106]: Also not matching the actual arguments (missing `detail_level`)

File: indico/modules/events/sessions/templates/display/session_display.html (hunk 6 above)
Comment (javfg) [reply indico/modules/events/sessions/templates/display/session_display.html:67]: If I'm not mistaken, the same check and the same element is added in lines 58-65:

https://github.com/indico/indico/blob/749bbc98f82251ef874c43e799509e201f67dc88/indico/modules/events/sessions/templates/display/session_display.html#L58-L65

I removed this because the final html ended up with two divs with that id. Am I missing something?

Comment (ThiefMaster) [reply indico/modules/events/sessions/templates/display/session_display.html:67]: Oh right. I didn't notice the `start_dt` check being around the other code as well.

File: indico/modules/events/ical.py (hunk 8 above)
Comment (javfg) [reply indico/modules/events/ical.py:56]: Actually, the original comment is incorrect. It is sessions which don't have `person_links`, but they do not have `speakers` either, so I think removing the elif could also work.

File: indico/modules/events/static/offline.py (hunk 7 above)
Comment (javfg) [reply indico/modules/events/static/offline.py:370]: Tested, it works!

File: indico/modules/events/ical.py [hunk 10]
```diff
@@ -12,72 +12,125 @@
...
+        contact_info = f'{event.contact_title}'
+        if len(event.contact_emails):
+            contact_info += f'; {"; ".join(event.contact_emails)}'
+        if len(event.contact_phones):
+            contact_info += f'; {"; ".join(event.contact_phones)}'
+        component.add('contact', contact_info)
+
+    # add logo url if event is public
+    if event.effective_protection_mode == ProtectionMode.public and event.has_logo:
+        component.add('image', f"{config.BASE_URL}{event.logo_url}", {'VALUE': 'URI'})
```
Comment (ThiefMaster) [inline indico/modules/events/ical.py:87]:
```suggestion
        component.add('image', event.external_logo_url, {'VALUE': 'URI'})
```

File: indico/modules/events/ical.py [hunk 11]
```diff
@@ -12,72 +12,125 @@
...
+    """Generates an event icalendar component from an Indico event."""
+
+    uid = f'indico-event-{event.id}@{url_parse(config.BASE_URL).host}'
+    component = generate_basic_component(event, uid)
+
+    # add contact title, phones and emails if present
+    if event.contact_title:
+        contact_info = f'{event.contact_title}'
+        if len(event.contact_emails):
+            contact_info += f'; {"; ".join(event.contact_emails)}'
```
Comment (ThiefMaster) [inline indico/modules/events/ical.py:80]: I'd really avoid nesting quoted strings inside f-strings. In this particular case I'd make `contact_info` a list and append those segments and then join them with `;` when adding the contact component.

File: CHANGES.rst [hunk 12]
```diff
@@ -40,6 +40,8 @@ Improvements
   (:pr:`4817`)
 - Failed login attempts now trigger rate limiting to prevent brute-force attacks
   (:issue:`1550`, :pr:`4817`)
+- iCalendar exports include more information and use new serializers
```
Comment (ThiefMaster) [inline CHANGES.rst:43]: I'd reword the changelog entry a bit. "and use new serializers" is not relevant for the changelog (that's purely internal refactoring), and I'd mention at least slightly more details on what the actual (user-facing) improvements to the ical exports are

File: indico/modules/events/contributions/ical.py [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+from indico.modules.events.ical import generate_basic_component
+from indico.web.flask.util import url_for
+
+
+def generate_contribution_component(contribution, related_to_uid=None):
+    """Generates an Event icalendar component from an Indico Contribution.
+
+    :param contribution: The Indico Contribution to use
+    :param related_to_uid: Indico uid used in related_to field
+    :returns: an icalendar Event
```
Comment (ThiefMaster) [inline indico/modules/events/contributions/ical.py:21]: Usually we use `:return:` (16 vs 117 results)

File: indico/modules/events/sessions/controllers/display.py [hunk 14]
```diff
@@ -72,7 +73,10 @@ def _process(self):
 @allow_signed_url
 class RHExportSessionToICAL(RHDisplaySessionBase):
     def _process(self):
-        return send_file('session.ics', get_session_ical_file(self.session), 'text/calendar')
+        detailed = request.args.get('detail', 'sessions') == 'contributions'
```
Comment (ThiefMaster) [inline indico/modules/events/sessions/controllers/display.py:76]:
```suggestion
        detailed = request.args.get('detail') == 'contributions'
```

The default value is no longer needed

File: indico/modules/events/sessions/ical.py [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+from icalendar.cal import Calendar
+from werkzeug.urls import url_parse
+
+from indico.core.config import config
+from indico.modules.events.ical import generate_basic_component
+from indico.web.flask.util import url_for
+
+
+def generate_session_component(session, related_to_uid=None):
+    """Generates an Event iCalendar component from an Indico Session."""
```
Comment (ThiefMaster) [inline indico/modules/events/sessions/ical.py:17]:
```suggestion
    """Generate an Event iCalendar component from an Indico Session."""
```

> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".

We fixed many of docstrings in #4649 to follow [PEP-257](https://www.python.org/dev/peps/pep-0257/) so let's keep new ones consistent :)

File: indico/modules/events/sessions/ical.py [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+    return component
+
+
+def session_to_ical(session, detailed=False):
+    """Serialize a session into an iCal.
+
+    :param session: The session to serialize
+    :param detailed: If True, iCal will include the session's contributions
+    """
+
```
Comment (ThiefMaster) [inline indico/modules/events/sessions/ical.py:35]: Only classes have a blank link between docstring and code

File: indico/modules/events/templates/header.html [hunk 17]
```diff
@@ -201,7 +201,8 @@
                     {{ _render_filters() }}
                 {% endif %}
 
-                <div id="event-calendar-link" data-event-id="{{ event.id }}"></div>
+                <div id="event-calendar-link" data-event-id="{{ event.id }}"
+                     data-event-contrib-count={{ event.contributions|length }}></div>
```
Comment (ThiefMaster) [inline indico/modules/events/templates/header.html:205]:
```suggestion
                     data-event-contrib-count="{{ event.contributions|length }}"></div>
```

File: indico/modules/events/controllers/display.py [hunk 18]
```diff
@@ -9,36 +9,24 @@
...
-        for update in values_from_signal(
-            signals.event.metadata_postprocess.send('ical-export', event=self.event, data=data, user=session.user),
-            as_list=True
-        ):
-            data.update(update)
-
-        response = {'results': data}
-        serializer = Serializer.create('ics')
-        return send_file('event.ics', BytesIO(serializer(response)), 'text/calendar')
+        detailed = request.args.get('detail', 'events') == 'contributions'
```
Comment (ThiefMaster) [inline indico/modules/events/controllers/display.py:27]:
```suggestion
        detailed = request.args.get('detail') == 'contributions'
```

File: CHANGES.rst [hunk 19]
```diff
@@ -40,6 +40,10 @@ Improvements
   (:pr:`4817`)
 - Failed login attempts now trigger rate limiting to prevent brute-force attacks
   (:issue:`1550`, :pr:`4817`)
+- iCalendar exports now include contact data, event logo URL and, when exporting
+  sessions/contributions, the UID of the related event. Also, only non-empty fields
+  are exported. (:issue:`4785`, :issue:`4586`, :issue:`4587`, :issue: `4791`,
```
Comment (ThiefMaster) [inline CHANGES.rst:45]:
```suggestion
  are exported. (:issue:`4785`, :issue:`4586`, :issue:`4587`, :issue:`4791`,
```

File: indico/modules/events/contributions/ical.py [hunk 20]
```diff
@@ -0,0 +1,48 @@
...
+
+
+def generate_contribution_component(contribution, related_to_uid=None):
+    """Generate an Event icalendar component from an Indico Contribution.
+
+    :param contribution: The Indico Contribution to use
+    :param related_to_uid: Indico uid used in related_to field
+    :return: an icalendar Event
+    """
+    uid = 'indico-contribution-{}@{}'.format(contribution.id, url_parse(config.BASE_URL).host)
```
Comment (ThiefMaster) [inline indico/modules/events/contributions/ical.py:23]:
```suggestion
    uid = f'indico-contribution-{contribution.id}@{url_parse(config.BASE_URL).host}'
```

File: indico/modules/events/contributions/ical.py [hunk 21]
```diff
@@ -0,0 +1,48 @@
...
+def contribution_to_ical(contribution):
+    """Serialize a contribution into an ical.
+
+    :param contribution: The contribution to serialize
+    """
+    calendar = Calendar()
+    calendar.add('version', '2.0')
+    calendar.add('prodid', '-//CERN//INDICO//EN')
+
+    related_event_uid = 'indico-event-{}@{}'.format(contribution.event.id, url_parse(config.BASE_URL).host)
```
Comment (ThiefMaster) [inline indico/modules/events/contributions/ical.py:42]:
```suggestion
    related_event_uid = f'indico-event-{contribution.event.id}@{url_parse(config.BASE_URL).host}'
```

File: indico/modules/events/ical.py [hunk 22]
```diff
@@ -12,72 +12,121 @@
...
+def generate_basic_component(entity, uid=None, url=None):
+    """Generate an iCalendar component with basic common properties.
+
+    :param entity: Event/session/contribution where properties come from
+    :param uid: UID for the component
+    :param url: URL for the component (defaults to `entity.external_url`)
+
+    :return: iCalendar event with basic properties
+    """
+    component = Event()
```
Comment (ThiefMaster) [inline indico/modules/events/ical.py:29]: I would prefer `icalendar.Event()` here to be more clear that it's not an indico `Event` instance

File: indico/modules/events/ical.py [hunk 23]
```diff
@@ -12,72 +12,121 @@
...
+                    for x in speaker_list]
+        description.append('Speakers: {}'.format(', '.join(speakers)))
+    if entity.description:
+        desc_text = str(entity.description) or '<p/>'  # get rid of RichMarkup
+        try:
+            description.append(str(html.fromstring(desc_text).text_content()))
+        except ParserError:
+            # this happens if desc_text only contains a html comment
+            pass
+    if len(description):
```
Comment (ThiefMaster) [inline indico/modules/events/ical.py:63]:
```suggestion
    if description:
```

this should be enough; empty lists are falsy

File: indico/modules/events/ical.py [hunk 24]
```diff
@@ -12,72 +12,121 @@
...
+    uid = f'indico-event-{event.id}@{url_parse(config.BASE_URL).host}'
+    component = generate_basic_component(event, uid)
+
+    # add contact title, phones and emails if present
+    contact_info = [event.contact_title]
+    if event.contact_title:
+        if len(event.contact_emails):
+            contact_info += event.contact_emails
+        if len(event.contact_phones):
+            contact_info += event.contact_phones
```
Comment (ThiefMaster) [inline indico/modules/events/ical.py:80]:
```suggestion
        contact_info += event.contact_emails
        contact_info += event.contact_phones
```

appending an empty list is a noop, so no need to check if it's empty first

File: indico/modules/events/ical.py [hunk 25]
```diff
@@ -12,72 +12,121 @@
...
+
+
+def generate_event_component(event):
+    """Generates an event icalendar component from an Indico event."""
+    uid = f'indico-event-{event.id}@{url_parse(config.BASE_URL).host}'
+    component = generate_basic_component(event, uid)
+
+    # add contact title, phones and emails if present
+    contact_info = [event.contact_title]
+    if event.contact_title:
```
Comment (ThiefMaster) [inline indico/modules/events/ical.py:76]: I don't think this is ever empty since that field is required (and if it could be empty, the line before would need a check for it)

File: indico/modules/events/contributions/ical.py [hunk 26]
```diff
@@ -0,0 +1,48 @@
...
+def contribution_to_ical(contribution):
+    """Serialize a contribution into an ical.
+
+    :param contribution: The contribution to serialize
+    """
+    calendar = icalendar.Calendar()
+    calendar.add('version', '2.0')
+    calendar.add('prodid', '-//CERN//INDICO//EN')
+
+    related_event_uid = 'indico-event-{contribution.event.id}@{url_parse(config.BASE_URL).host}'
```
Comment (ThiefMaster) [inline indico/modules/events/contributions/ical.py:42]:
```suggestion
    related_event_uid = f'indico-event-{contribution.event.id}@{url_parse(config.BASE_URL).host}'
```

File: indico/modules/events/ical.py [hunk 27]
```diff
@@ -5,79 +5,125 @@
...
+                calendar.add_component(component)
+
+    data = calendar.to_ical()
+
+    # check whether the plugins want to add/override any data
+    for update in values_from_signal(
+        signals.event.metadata_postprocess.send('ical-export', event=event, data=data, user=user),
+        as_list=True
+    ):
+        data.update(update)
```
Comment (ThiefMaster) [inline indico/modules/events/ical.py:127]: This doesn't look correct. `to_ical()` already returns a string, but the signal expects a dict. Since this is only for events, it should probably be either in `generate_event_component` or `generate_basic_component`...

File: indico/modules/events/contributions/ical.py [hunk 28]
```diff
@@ -0,0 +1,48 @@
...
+    component = generate_contribution_component(contribution, related_event_uid)
+    calendar.add_component(component)
+
+    data = calendar.to_ical()
+
+    return data
```
Comment (ThiefMaster) [inline indico/modules/events/contributions/ical.py:48]:
```suggestion
    return calendar.to_ical()
```

File: indico/modules/events/ical.py [hunk 29]
```diff
@@ -5,79 +5,125 @@
...
+def generate_event_component(event, user=None):
+    """Generate an event icalendar component from an Indico event."""
+    uid = f'indico-event-{event.id}@{url_parse(config.BASE_URL).host}'
+    component = generate_basic_component(event, uid)
+
+    # add contact title, phones and emails
+    contact_info = [event.contact_title]
+    contact_info += event.contact_emails
+    contact_info += event.contact_phones
+    component.add('contact', ';'.join(contact_info))
```
Comment (ThiefMaster) [inline indico/modules/events/ical.py:78]: When testing this I realized that including the title is pretty useless - it's just "Support" or "Contact" or something like that.

```suggestion
    # add contact information
    contact_info = event.contact_emails + event.contact_phones
    if contact_info:
        component.add('contact', ';'.join(contact_info))
```

File: indico/modules/events/ical.py [hunk 30]
```diff
@@ -5,79 +5,125 @@
...
-
-        calendar.add_component(cal_event)
+        if not detailed:
+            component = generate_event_component(event, user)
+            calendar.add_component(component)
+        else:
+            from indico.modules.events.contributions.ical import generate_contribution_component
+            components = [
+                generate_contribution_component(contribution) for contribution in event.contributions
+            ]
```
Comment (ThiefMaster) [inline indico/modules/events/ical.py:125]:
```suggestion
            components = [
                generate_contribution_component(contrib)
                for contrib in event.contributions
                if contrib.start_dt
            ]
```

otherwise it fails if an event has unscheduled contributions

**Review by ThiefMaster — APPROVED** [review]
🎉

File: indico/modules/events/sessions/ical.py [hunk 31]
```diff
@@ -0,0 +1,50 @@
...
+def generate_session_component(session, related_to_uid=None):
+    """Generate an Event iCalendar component from an Indico Session."""
+    uid = f'indico-session-{session.id}@{url_parse(config.BASE_URL).host}'
+    url = url_for('sessions.display_session', session, _external=True)
+    component = generate_basic_component(session, uid, url)
+
+    if related_to_uid:
+        component.add('related_to', related_to_uid)
+
```
Comment (dilyanpalauzov) [inline indico/modules/events/sessions/ical.py:24]: This line is suspicious.  I do not say that the code is wrong, I have not run it.

The property is officially called with hyphen `RELATED-TO`, but the code adds a property with underscore `related_to`.

Comment (javfg) [reply indico/modules/events/sessions/ical.py:24]: Good catch, fix on the way.
